factory/usr/lib/systemd/system/initrd-switch-root.service.d/core-override.conf

```diff
@@ -1,18 +0,0 @@
-[Service]
-# This empty "ExecStart" does override the original ExecStat from the
-# "real" initrd-switch-root.service that would do a stateful re-exec.
-ExecStart=
-# Currently doing a stateful re-exec will fail. The reasons are a bit
-# unclear, it seems to be related to the fact that systemd is confused
-# by the mount units in the initrd that are not there in the new pivot
-# root. So this systemctl call to "/sbin/init" is a workaround to prevent
-# systemd from doing a stateful re-exec (there is no other way to
-# prevent this).
-#
-# Using "/sbin/init" here directly also ensures that the kernel commandline
-# "init=" parameter is ignored as it would allow to bypass our protections.
-#
-# XXX: we should fix this eventually (this may involve upstream work)
-# because this also prevents "systemd-analyze time" from displaying
-# the time spend in the initrd.
-ExecStart=/bin/systemctl --no-block switch-root /sysroot /sbin/init
```

Comment:
Removing this file, will result in serialized re-exec from initrd to rootfs.  But it will also make `init=` argument on the command line work as well. which I find a bit scary. I hope that snapd prevents/prohibits sealing against customized `init=` arguments cause that would expose encryption keys in the root shell without any authentication.

@mvo5 @pedronis @chrisccoulson

Reply:
yes you cannot change kernel command line in a system with full disk encryption unless you control the gadget snap and can upload / install a new gadget snap with a new cmdline.full or cmdline.extra file, in which case we re-seal against the new kernel command line. So just booting with a init= ... whatever will not expose anything since the encrypted partition won't be able to be unlocked.